Doc/library/zipfile.rst

```diff
@@ -637,22 +641,41 @@ file:
    .. versionadded:: 3.8
       The *strict_timestamps* keyword-only argument
 
+   .. versionadded:: 3.9
+      The *follow_symlinks* keyword-only argument
+
 
 Instances have the following methods and attributes:
 
 .. method:: ZipInfo.is_dir()
 
-   Return ``True`` if this archive member is a directory.
+   Return :const:`True` if this archive member is a directory.
 
+   .. versionadded:: 3.6
    This uses the entry's name: directories should always end with ``/``.
 
-   .. versionadded:: 3.6
+   .. versionchanged:: 3.9
+   Use the filemode contained in external file attributes.
+
+.. method:: ZipInfo.is_reg()
 
+   Return :const:`True` if this archive member is a regular file.
+
+.. method:: ZipInfo.is_file()
+
+   Same as :meth:`is_reg`.
+
+.. method:: ZipInfo.is_sym()
```

Comment:
Is there a reason why you chose `is_sym` as opposed to `issym()` (which would mirror `TarInfo.issym()`)? https://docs.python.org/3/library/tarfile.html#tarfile.TarInfo.issym

Reply:
I actually wanted to mirror `issym()` but there was a `is_dir()` already present, and I should have renamed it (which would be sad for backward compatibility I guess). But we can either :

1. keep `is_thing()`
2. use `isthing()` and redirect `is_dir()` to `isdir()`
3. use `isthing()` and warn users of the removal of `is_dir()`

Reply:
Ok, I think 1. is probably the best, given that PEP 8 suggests to separate function names with underscores.